### PR TITLE
Converts user-facing documentation to GroovyDoc.

### LINKED ISSUE #141: Convert J2objcPluginExtension comments to groovydoc
@brunobowden we should be able to convert our comments to groovydoc and then publish a description of our DSL automatically.  Or do you want to duplicate all this info in README.md and/or in J2objcPlugin.groovy?  I prefer not duplicating.  When we publish a new version, one of the steps would be publishing the groovydoc to the github.com project site.  How does that sound?

### PR DESCRIPTION
**Pull request by advayDev1**
Fixes #141

### CODE REVIEW

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPluginExtension.groovy [hunk 1]
```diff
@@ -20,50 +20,79 @@ import groovy.transform.PackageScope
...
+     */
     String[] generatedSourceDirs = []
 
-
     // CYCLEFINDER
+    /**
+     * Whether to skip the j2objcCycleFinder task.
+     *
+     * @deprecated Prefer excluding the task in your own build.gradle using
+     * {@code j2objcCycleFinder &#123; enabled = false &#125; }
```
Comment (brunobowden) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPluginExtension.groovy]: We'd like the task to be disabled by default and then enabled if someone chooses. If you can, make that change in another commit.

Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPluginExtension.groovy]: Will do.

**Comment by confile** [comment]
LGTM

**Comment by advayDev1** [comment]
thanks @confile ;

@brunobowden - all good?

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPluginExtension.groovy [hunk 2]
```diff
@@ -87,16 +114,35 @@ class J2objcPluginExtension {
...
+    boolean filenameCollisionCheck = true
+
+    /**
+     * Filter on files to translate.
+     * <p/>
+     * If no pattern is specified, all files within the sourceSets are translated.
+     * <p/>
+     * This filter is applied on top of all files within the 'main' and 'test'
+     * java sourceSets.  Use {@link #translatePattern(groovy.lang.Closure)} to
+     * configure.
```
Comment (brunobowden) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPluginExtension.groovy]: I find the concrete example much easier to work. Also include reference to where they can read more:
```
translatePattern {
    exclude 'CannotTranslateFile.java'
    exclude '**/CannotTranslateDir/*.java'
    include '**/CannotTranslateDir/AnExceptionToInclude.java'
}
// More Info: https://docs.gradle.org/current/userguide/working_with_files.html#sec:file_trees
```

Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPluginExtension.groovy]: the example is on the DSL method because you can't do that with the field/property directly.  but I will make that example fuller.

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPluginExtension.groovy [hunk 3]
```diff
@@ -104,32 +150,53 @@ class J2objcPluginExtension {
...
+    /**
+     * Flags copied verbatim to testrunner command.
+     */
     String testFlags = ""
-    // Error if it runs less than the expected number of tests, set to 0 to disable.
-    // It is set as a minimum so adding a unit test doesn't break the j2objc build.
+    /**
+     * j2objcTest will fail if it runs less than the expected number of tests; set to 0 to disable.
+     * <p/>
+     * It is set as a minimum so adding a unit test doesn't break the j2objc build.
```
Comment (brunobowden) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPluginExtension.groovy]: Remove `set as`:

`It is a minimum...`

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPluginExtension.groovy (hunk 2 above)
Comment (brunobowden) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPluginExtension.groovy]: Add a another comment explaining the distinction between the two and how they interact.

For testPattern, either duplicate the example or have it link to translatePattern documentation

**Comment by brunobowden** [comment]
Looks good after you respond to my changes

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPluginExtension.groovy (hunk 2 above)
Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPluginExtension.groovy]: done for testPattern.  but i'm not sure what 'the two' refers to?  the test vs. translate?  or the field vs. the DSL method?

Comment (brunobowden) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPluginExtension.groovy]: field vs. DSL method

**Comment by brunobowden** [comment]
looks good

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPluginExtension.groovy (hunk 2 above)
Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPluginExtension.groovy]: ah ok did that
